src/lightning/fabric/strategies/deepspeed.py

```diff
@@ -507,7 +507,11 @@ def load_checkpoint(
 
         optimzer_state_requested = any(isinstance(item, (Optimizer, DeepSpeedOptimizer)) for item in state.values())
 
-        torch.cuda.empty_cache()
+        if isinstance(self.accelerator, Accelerator) and self.accelerator.get_device() != "cpu":
+            getattr(torch, self.root_device.type.split(":")[0]).empty_cache()
```

Comment:
Same goes for the comment above. `empty_cache` is not part of a contract (nor is the fact that a device is registered as a submodule of the `torch` module). It needs to be if we want to rely on calling `emtpy_cache` on whatever device we pass.

BTW, being a `torch` submodule is too strong of a requirement in my opinion.

In this case we should probably guard the strategy to be GPU-only.

Reply:
Sorry, I may not understand what you mean. Could you tell me more about your idea?